src/db/spanner/models.rs

```diff
@@ -1371,9 +1314,11 @@ impl SpannerDb {
         // https://bugzilla.mozilla.org/show_bug.cgi?id=963332
 
         let next_offset = if limit >= 0 && bsos.len() > limit as usize {
-            bsos.pop();
-            let modifieds: Vec<i64> = bsos.iter().map(|r| r.modified.as_i64()).collect();
-            self.encode_next_offset(sort, offset, timestamp.map(|t| t.as_i64()), modifieds)
+            let bso = bsos.pop();
+            let (offset, timestamp) = bso
+                .map(|b| (Some(b.id), Some(b.modified.as_i64())))
+                .unwrap_or((None, None));
```

Comment:
Does this unwrap multiple values in a tuple? Cool.

Reply:
Yeah, it's a cheesy hack, but it works.